py-polars/tests/unit/datatypes/test_categorical.py

```diff
@@ -977,3 +977,31 @@ def test_categorical_prefill() -> None:
     s = pl.Series(["1", "2", "3"], dtype=pl.Categorical)
     s = s.filter([True, False, True])
     assert s.n_unique() == 2
+
+
+@pytest.mark.may_fail_auto_streaming  # not implemented
+def test_categorical_min_max() -> None:
+    lf = pl.LazyFrame(
+        {
+            "a": ["foo", "bar"],
+            "b": ["foo", "bar"],
+            "c": ["foo", "bar"],
+        },
+        schema={
+            "a": pl.Categorical("physical"),
+            "b": pl.Categorical("lexical"),
+            "c": pl.Enum(["foo", "bar"]),
+        },
+    )
+
+    q = lf.select(pl.all().min())
+    result = q.collect()
+    assert q.collect_schema() == lf.collect_schema()
+    assert result.schema == lf.collect_schema()
```

Comment:
Can we also check that the schema contains categoricals and enums?

Reply:
added more explicit check